integration-tests/webhook/webhook_test.go

```diff
@@ -0,0 +1,163 @@
+package webhook
+
+import (
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openebs/maya/integration-tests/artifacts"
+	k8s "github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	cvr "github.com/openebs/maya/pkg/cstorvolumereplica/v1alpha1"
+	pvc "github.com/openebs/maya/pkg/kubernetes/persistentvolumeclaim/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// stsYaml holds the yaml spec
+	// for statefulset application
+	clonePVCYaml artifacts.Artifact = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: demo-snap-vol-claim
+  annotations:
+    snapshot.alpha.kubernetes.io/snapshot: fastfurious
+spec:
+  storageClassName: openebs-snapshot-promoter
+  accessModes: [ "ReadWriteOnce" ]
+  resources:
+    requests:
+      storage: 2G`
+)
+
+var _ = Describe("AdmissionWebhook", func() {
+	BeforeEach(func() {
+		// Extracting storageclass artifacts unstructured
+		_, err := artifacts.GetArtifactUnstructured(
+			artifacts.Artifact(artifacts.SingleReplicaSC),
+		)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting PVC artifacts unstructured
+		PVCUnst, err := artifacts.GetArtifactUnstructured(artifacts.Artifact(artifacts.CStorPVCArtifacts))
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Extracting statefulset application namespace
+		PVCNamespace := PVCUnst.GetNamespace()
+
+		// Webhook stuffs
+		client, err := k8s.Clientset().Get()
+		client.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().List(metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			ginkgo.Skip("dynamic configuration of webhooks requires the admissionregistration.k8s.io group to be enabled")
+		}
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Create pvc using storageclass 'cstor-sparse-pool'
+		cu := k8s.CreateOrUpdate(
+			k8s.GroupVersionResourceFromGVK(PVCUnst),
+			PVCUnst.GetNamespace(),
+		)
+		_, err = cu.Apply(PVCUnst)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		// Check for pvc to get created and bound
+		Eventually(func() string {
+			pvclaim, err := pvc.
+				KubeClient(pvc.WithNamespace(PVCNamespace)).Get(PVCNamespace, PVCUnst.GetName())
+			Expect(err).ShouldNot(HaveOccurred())
+			return string(pvclaim.Status.Phase)
+		},
+			defaultTimeOut, defaultPollingInterval).
+			Should(Receive((ContainSubstring("Bound")), "Pvc phase should bound"))
+
+		PVLabel := "openebs.io/casType=" + "cstor"
```

Comment:
if this test suite is specific to cstor then this test case should be tagged as `[cstor]`
Let this constant be a global constant

Reply:
We will have these kind of tags in coming PR's